### DISCUSSION TITLE
Updating Dashboard via API

### DISCUSSION THREAD

**Discussion by lucaschernbeck** [getredash/redash#6361] [Q&A]
Hi all, I've been trying to update a dashboard in my Redash instance via the [refresh_dashboard.py](https://github.com/getredash/redash-toolbelt/blob/master/redash_toolbelt/examples/refresh_dashboard.py) python script.
While the terminal returns 200s for each of the queries on that dashboard the data doesn't actually seem to update in the UI. Additionally, the "last update" stamp at the bottom left of each widget also does not seem to update.
Am I doing something wrong?

**Comment by justinclift** [comment]
It might be worth watching the Redash logs from the command line as you run the script, to see if any weirdness / errors / etc are displayed.

How familiar are you with using docker and/or the command line?  This approach is probably mostly suitable for people ok with using a command line interface though.  (mentioning just in case)

**Reply by lucaschernbeck** [reply]
Thank you for your reply @justinclift.
I'm somewhat familiar with command line but docker no so much I'm afraid.

**Reply by justinclift** [reply]
No worries.  Being familiar with Docker would make things a lot easier, but there are workarounds up to a point. :smile:

For your situation, I'd probably try and capture the Docker log file output using the process here:

https://github.com/getredash/redash/discussions/6128#discussioncomment-6302843

That pretty much dumps the entire Docker log output into one big file, in time based order.  So, most recent stuff is at the end.  Note that the file *can* be pretty big for some installations, so just be aware. :wink:

Anyway, if you run the python script then capture the log file, check the time/date stamp range for when the script was running.  See if anything weird or unexpected shows up there.  In theory (!), it should have some info about what's happening. :smile:
